### PR TITLE
Remove redundant if statements

### PR DESCRIPTION
**Pull request by DanVanAtta**
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/782)

### CODE REVIEW

File: src/games/strategy/triplea/delegate/GameStepPropertiesHelper.java [hunk 1]
```diff
@@ -71,11 +71,8 @@ public static boolean isAirborneMove(final GameData data) {
       final String prop = data.getSequence().getStep().getProperties().getProperty(GameStep.PROPERTY_airborneMove);
       if (prop != null) {
         isAirborneMove = Boolean.parseBoolean(prop);
-      } else if (isAirborneDelegate(data)) {
-        isAirborneMove = true;
-      } else {
-        isAirborneMove = false;
-      }
+      } else
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/delegate/GameStepPropertiesHelper.java]: Missing bracket.

File: src/games/strategy/triplea/delegate/GameStepPropertiesHelper.java [hunk 2]
```diff
@@ -144,16 +141,9 @@ public static boolean isFireRockets(final GameData data) {
...
-        } else {
-          isFireRockets = false;
-        }
-      } else if (isNonCombatDelegate(data)) {
-        isFireRockets = true;
-      } else {
-        isFireRockets = false;
-      }
+        isFireRockets = isCombatDelegate(data);
+      } else
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/delegate/GameStepPropertiesHelper.java]: Missing bracket.

File: src/games/strategy/triplea/delegate/GameStepPropertiesHelper.java [hunk 3]
```diff
@@ -179,11 +169,8 @@ public static boolean isRepairUnits(final GameData data) {
           isRepairUnits = Boolean.parseBoolean(prop);
         } else if (isCombatDelegate(data) && repairAtStartAndOnlyOwn) {
           isRepairUnits = true;
-        } else if (data.getSequence().getStep().getName().endsWith("EndTurn") && repairAtEndAndAll) {
-          isRepairUnits = true;
-        } else {
-          isRepairUnits = false;
-        }
+        } else
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/delegate/GameStepPropertiesHelper.java]: Missing bracket.

File: src/games/strategy/triplea/delegate/GameStepPropertiesHelper.java [hunk 4]
```diff
@@ -201,11 +188,8 @@ public static boolean isGiveBonusMovement(final GameData data) {
       final String prop = data.getSequence().getStep().getProperties().getProperty(GameStep.PROPERTY_giveBonusMovement);
       if (prop != null) {
         isBonus = Boolean.parseBoolean(prop);
-      } else if (isCombatDelegate(data)) {
-        isBonus = true;
-      } else {
-        isBonus = false;
-      }
+      } else
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/delegate/GameStepPropertiesHelper.java]: Missing bracket.

File: src/games/strategy/triplea/delegate/GameStepPropertiesHelper.java [hunk 5]
```diff
@@ -229,11 +213,8 @@ public static boolean isRemoveAirThatCanNotLand(final GameData data) {
         isRemoveAir = false;
       } else if (isNonCombatDelegate(data)) {
         isRemoveAir = true;
-      } else if (data.getSequence().getStep().getName().endsWith("Place")) {
-        isRemoveAir = true;
-      } else {
-        isRemoveAir = false;
-      }
+      } else
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/delegate/GameStepPropertiesHelper.java]: Missing bracket.

File: src/games/strategy/triplea/delegate/GameStepPropertiesHelper.java [hunk 6]
```diff
@@ -307,11 +288,8 @@ public static boolean isResetUnitStateAtEnd(final GameData data) {
           data.getSequence().getStep().getProperties().getProperty(GameStep.PROPERTY_resetUnitStateAtEnd);
       if (prop != null) {
         isReset = Boolean.parseBoolean(prop);
-      } else if (isNonCombatDelegate(data)) {
-        isReset = true;
-      } else {
-        isReset = false;
-      }
+      } else
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/delegate/GameStepPropertiesHelper.java]: Missing bracket.

File: src/games/strategy/triplea/delegate/GameStepPropertiesHelper.java [hunk 7]
```diff
@@ -327,11 +305,8 @@ public static boolean isBid(final GameData data) {
         isBid = Boolean.parseBoolean(prop);
       } else if (isBidPurchaseDelegate(data)) {
         isBid = true;
-      } else if (isBidPlaceDelegate(data)) {
-        isBid = true;
-      } else {
-        isBid = false;
-      }
+      } else
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/delegate/GameStepPropertiesHelper.java]: Missing bracket.

File: src/games/strategy/triplea/delegate/Matches.java [hunk 8]
```diff
@@ -135,11 +135,8 @@ public boolean match(final Unit unit) {
       final UnitAttachment ua = UnitAttachment.get(unit.getType());
       if (ua.getTransportCapacity() == -1) {
         return true;
-      } else if (ua.getIsCombatTransport() && ua.getIsSea()) {
-        return true;
-      } else {
-        return false;
-      }
+      } else
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/delegate/Matches.java]: Missing bracket.

**Comment by RoiEXLab** [comment]
@ron-murhammer about the Missing Bracket thing... I think you're wrong...In most cases I'm seeing 5 brackets being removed and 1 being added...

**Comment by DanVanAtta** [comment]
PR updated, merge conflicts resolved.

@RoiEXLab  - it does appear the auto-simplify done by intelliJ on this one did use simplified 'else' blocks, that did not have the curly braces.

I added in the curly braces, but also there was a lot of opportunity to simplify the if/else blocks.

**Comment by RoiEXLab** [comment]
@DanVanAtta Time to re-self-assign...
boolean isFireRocktes in GameStepPropertiesHelper.java needs initialization

**Comment by RoiEXLab** [comment]
@DanVanAtta Uh oh. Seems like you broke it ^^
A test is failing...

**Comment by DanVanAtta** [comment]
@RoiEXLab  - I"m getting this on the latest master. Would you mind taking a look to see if you can repro?

```
games.strategy.engine.lobby.server.login.LobbyLoginValidatorTest > testAnonymousLoginBadName FAILED
    java.lang.IllegalArgumentException: You have an invalid MAC address!
        at games.strategy.net.MacFinder.getHashedMacAddress(MacFinder.java:24)
        at games.strategy.engine.lobby.server.login.LobbyLoginValidatorTest.testAnonymousLoginBadName(LobbyLoginValidatorTest.java:70)
```

**Comment by RoiEXLab** [comment]
@DanVanAtta Although it's 1 am here, and I was just about going to sleep, I took a look at this one...
I cannot reproduce this Test-Failure on master...
When testing your branch, I get the following failures/errors:
[Gist](https://gist.github.com/RoiEXLab/8b2014e7cf2261053a337edc9926e414)

**Comment by DanVanAtta** [comment]
@ron-murhammer  this is clean now. The refactor in this PR broke some tests, and I ran it on a machine that revealed additional test failures when running tests on a laptop via wifi. Both issues have been addressed, and for this PR - I removed the bad refactor and added instead a simple commit that added in curly braces

File: src/games/strategy/triplea/delegate/Matches.java (hunk 8 above)
Comment (ron-murhammer) [inline src/games/strategy/triplea/delegate/Matches.java]: This bracket still appears to be missing.

**Comment by ron-murhammer** [comment]
@DanVanAtta Looks good but I think you missed 1 bracket :)

**Comment by DanVanAtta** [comment]
fixed it (c70afa2) - now updated

**Pull request merged**